plugins/book-data-extension/src/plugin/index.ts

```diff
@@ -1,2 +1,49 @@
-export const dunnystring =
-	"This is a dummy string so that the 'build' script doesn't fail (warn = fail in CI) due to empty bundle. Remove when there's something to build.";
+import { BookFetcherPlugin as BookFetcherPluginType, BookEntry } from "@librocco/db";
+import { postMessage, addEventListener, removeEventListener } from "./window-helpers";
+import { listenForBooks } from "./listeners";
+
+export const BookFetcherPlugin = (): BookFetcherPluginType => {
+	const fetchBookData = async (isbns: string[]): Promise<BookEntry[]> => {
+		// Check if the extension is registered if not resolve to []
+
+		postMessage(`BOOK_FETCHER_REQ:checkForExtension:${isbns}`);
+
+		const extensionAvailable = await new Promise((resolve) => {
+			let resolved = false;
+
+			const promiseHandler = () => {
+				window.removeEventListener("message", handler);
+
+				if (!resolved) {
+					clearTimeout(promiseTimer);
+					resolve(false);
+				}
+			};
+			const promiseTimer = setTimeout(promiseHandler, 1000);
+
+			const handler = (event: MessageEvent) => {
+				if (event.source !== window) return;
+				if (event.data.message && event.data.message === `BOOK_FETCHER_REQ:replyFromExtension:`) {
```

Comment:
Wouldn't it make more sense to make this `BOOK_FETCHER:RES` (all intention is conveyed in a much shorter and consistently formatted string)